src/ucp/core/ucp_request.c

```diff
@@ -112,9 +113,9 @@ UCS_PROFILE_FUNC_VOID(ucp_request_cancel, (worker, request),
     if (req->flags & UCP_REQUEST_FLAG_EXPECTED) {
         UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
-        ucp_tag_exp_remove(&worker->tm, req);
+        removed = ucp_tag_exp_remove(&worker->tm, req);
         /* If tag posted to the transport need to wait its completion */
-        if (!(req->flags & UCP_REQUEST_FLAG_OFFLOADED)) {
+        if (removed && !(req->flags & UCP_REQUEST_FLAG_OFFLOADED)) {
```

Comment:
can remove `removed` variable and do:
```
if (ucp_tag_exp_remove(&worker->tm, req) &&
    !(req->flags & UCP_REQUEST_FLAG_OFFLOADED)) {
```
but it's probably bad for readability, wdyt?